File: lab02_stack/lab02_example01/scripts/publisher_node.py
#!/usr/bin/env python

import rospy
from std_msgs.msg import Float64MultiArray

def talker():

    pub = rospy.Publisher('chatter', Float64MultiArray, queue_size=10)
    rospy.init_node('talker', anonymous=True)
    rate = rospy.Rate(10) # 10hz
    count = 0

    num_msg = Float64MultiArray()
    num_msg.data = []

    while not rospy.is_shutdown():

        num_msg.data.append(count)

        pub.publish(num_msg)
        rate.sleep()
        count += 1

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass

File: lab02/lab02example01/src/publisher_node.cpp
#include "ros/ros.h"
#include "std_msgs/Float64MultiArray.h"

int main(int argc, char **argv)
{

    ros::init(argc, argv, "talker");

    ros::NodeHandle nh;

    ros::Publisher pub = nh.advertise<std_msgs::Float64MultiArray>("/chatter", 1000);

    ros::Rate loop_rate(10);
    std_msgs::Float64MultiArray msg;

    int count = 0;

    while (ros::ok())
    {


        msg.data.push_back(count);

        pub.publish(msg);

        ros::spinOnce();

        loop_rate.sleep();
        count++;
    }


    return 0;
}

